File: plugins/uptime/uptime.1m.js
#!/usr/bin/env node
// uptime.1m.js
const { execSync } = require('child_process');

function getUptime() {
    try {
        return execSync('crossbar uptime', { encoding: 'utf8', timeout: 5000 }).trim();
    } catch { return null; }
}

const uptime = getUptime();
if (uptime) {
    console.log(`⬆️ ${uptime}`);
    console.log("---");
    console.log(`System Uptime: ${uptime}`);
    console.log("Refresh | refresh=true");
} else {
    console.log("⬆️ --");
    console.log("---");
    console.log("Unable to get uptime");
}

File: plugins/uptime/uptime.1m.py
#!/usr/bin/env python3
# uptime.1m.py
import subprocess
import sys

def get_uptime():
    try:
        res = subprocess.run(['crossbar', 'uptime'], capture_output=True, text=True, timeout=5)
        if res.returncode == 0:
            return res.stdout.strip()
    except: pass
    return None

uptime = get_uptime()
if uptime:
    print(f"⬆️ {uptime}")
    print("---")
    print(f"System Uptime: {uptime}")
    print("Refresh | refresh=true")
else:
    print("⬆️ --")
    print("---")
    print("Unable to get uptime")